Makefile

```diff
@@ -37,6 +37,11 @@ BAZEL_STARTUP_ARGS := --client_debug $(BAZEL_STARTUP_ARGS)
 BAZEL_BUILD_ARGS := -s $(BAZEL_BUILD_ARGS)
 endif
 
+BUILD_WITH_CONTAINER ?= 0
```

Comment:
does this mean I do not need anything other than bazel installed locally?

Reply:
It's not different than just using bazel. Regular bazel uses sandboxes to run jobs, it's just not a docker sandbox. The problem would be that the generated build artifacts cannot be run locally since they link against container libraries. That would make any debugging or symbol resolution harder.

Reply:
Do keep in mind this warning: ```Note: Builds take noticeably more time to complete when the Docker sandbox feature is enabled. This is normal.``` 

We should probably enable RBE to utilize an entire cluster to run bazel, instead of using docker on a single machine.

Reply:
@mandarjog correct, but ideally Bazelisk instead of Bazel (to avoid updating it by hand).

@kyessenov it's different, since "sandboxed" builds still use the toolchain and build tools from the host environment, whereas docker build uses whatever is installed there.

I agree that we should use RBE (me & John are already sharing one build farm), but that's orthogonal to this change, which tries to remove local build dependencies.

Reply:
Well, there is a limited value to docker-sandbox. We already run bazel inside a docker container, so we don't need nested/peer containers. It would be nice to change Makefiles to use the build container to run everything.

Reply:
> We already run bazel inside a docker container, so we don't need nested/peer containers.

Yes, in the CI. The value of this is that developers don't need to have all the other build dependencies installed locally.